File: Scala/pi.scala
object pi extends App{
    def pi(n: Int): Double = {
        var num = 4.0
        var den = 1.0
        var oper = 1.0
        var pi = 0.0  
        var i = n
        while (i > 0){
            pi += oper * (num / den)
            den += 2.0
            oper *= -1.0
            i -= 1
            }
        return pi
    }
    println("Pi:" + pi(args(0).toInt))
}

File: Haskell/pi.hs
p = 0.0
num = 4.0
den = 1.0
oper = 1.0

pise :: Int -> Double -> Double -> Double -> Double -> Double
pise n p num den oper
    | n == 0 = p
    | otherwise = pise (n-1) (p + oper * (num / den)) num (den + 2) (oper * (-1))

main :: IO()
main = do
    print(pise 1000000 p num den oper)